#
# old_revision [baa27dc2b3071832da59aee6fbaceb7fa5fdf30e]
#
# patch "mtn-browse"
#  from [cef404cd565558fc86482c4d32ff1ca69c5ea04a]
#    to [3aefa00c204182ca76493989c0e1bd849e8ab02f]
# 
# patch "mtn-browse-Gtk2-SourceView2.patch"
#  from [4d0749c549610efb094a1813bb18cc2ec6103149]
#    to [da39a3ee5e6b4b0d3255bfef95601890afd80709]
#
============================================================
--- mtn-browse	cef404cd565558fc86482c4d32ff1ca69c5ea04a
+++ mtn-browse	3aefa00c204182ca76493989c0e1bd849e8ab02f
@@ -77,7 +77,7 @@ use Gtk2::Pango;
 use Gtk2::GladeXML;
 use Gtk2::Helper;
 use Gtk2::Pango;
-use Gtk2::SourceView;
+use Gtk2::SourceView2;
 use IO::Dir;
 use IO::File;
 use IPC::Open3;
@@ -2179,13 +2179,13 @@ sub get_browser_window(;$$$$$)
 
         # Setup the file file viewer (with syntax highlighting).
 
-        $browser->{file_view_svbuffer} = Gtk2::SourceView::Buffer->new(undef);
-        $browser->{file_view_svbuffer}->set_check_brackets(FALSE);
+        $browser->{file_view_svbuffer} = Gtk2::SourceView2::Buffer->new(undef);
+        $browser->{file_view_svbuffer}->set_highlight_matching_brackets(FALSE);
         $browser->{file_view_svbuffer}->set_max_undo_levels(0);
         $browser->{file_view_svbuffer}->begin_not_undoable_action();
         $browser->{file_view_svlangmgr} =
-            Gtk2::SourceView::LanguagesManager->new();
-        $browser->{file_view_sv} = Gtk2::SourceView::View->
+            Gtk2::SourceView2::LanguageManager->new();
+        $browser->{file_view_sv} = Gtk2::SourceView2::View->
             new_with_buffer($browser->{file_view_svbuffer});
         $browser->{file_view_sv}->modify_font($mono_font);
         $browser->{file_view_sv}->set_cursor_visible(FALSE);
@@ -3219,7 +3219,7 @@ sub update_browser_state($$)
             $browser->{file_view_svbuffer}->
                 place_cursor($browser->{file_view_svbuffer}->get_start_iter());
             $browser->{file_view_svbuffer}->set_text("");
-            $browser->{file_view_svbuffer}->set("highlight", FALSE);
+            $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
             set_label_value($browser->{file_name_value_label}, "");
             set_label_value($browser->{file_author_value_label}, "");
             set_label_value($browser->{last_update_value_label}, "");
@@ -3344,7 +3344,7 @@ sub display_file($$)
     $browser->{file_view_svbuffer}->
         place_cursor($browser->{file_view_svbuffer}->get_start_iter());
     $browser->{file_view_svbuffer}->set_text("");
-    $browser->{file_view_svbuffer}->set("highlight", FALSE);
+    $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
 
     # Get contents.
 
@@ -3448,9 +3448,12 @@ sub display_file($$)
                          && $mime_details->{syntax_highlight}))
                     && defined($lang =
                                $browser->{file_view_svlangmgr}->
-                                   get_language_from_mime_type($mime_type)))
+                                   guess_language($browser->
+                                                      {file_being_viewed}->
+                                                      {short_name},
+                                                  $mime_type)))
                 {
-                    $browser->{file_view_svbuffer}->set("highlight", TRUE);
+                    $browser->{file_view_svbuffer}->set_highlight_syntax(TRUE);
                     $browser->{file_view_svbuffer}->set_language($lang);
                 }
 
============================================================
--- mtn-browse-Gtk2-SourceView2.patch	4d0749c549610efb094a1813bb18cc2ec6103149
+++ mtn-browse-Gtk2-SourceView2.patch	da39a3ee5e6b4b0d3255bfef95601890afd80709
@@ -1,70 +0,0 @@
-#
-# old_revision [f08e10ab38607c498b1ccb41866ce82c1a52d37d]
-#
-# patch "mtn-browse"
-#  from [62051ca5d5c4130aeb6781d6907e1301ef10ccf3]
-#    to [9c5c4ca05dcacab4b2cf724da4b8293e324dc08f]
-#
-============================================================
---- mtn-browse	62051ca5d5c4130aeb6781d6907e1301ef10ccf3
-+++ mtn-browse	9c5c4ca05dcacab4b2cf724da4b8293e324dc08f
-@@ -76,7 +76,7 @@ use Gtk2::Pango;
- use Gtk2::GladeXML;
- use Gtk2::Helper;
- use Gtk2::Pango;
--use Gtk2::SourceView;
-+use Gtk2::SourceView2;
- use IO::Dir;
- use IO::File;
- use IPC::Open3;
-@@ -2236,13 +2236,13 @@ sub get_browser_window(;$$$$$)
- 
-         # Setup the file file viewer (with syntax highlighting).
- 
--        $browser->{file_view_svbuffer} = Gtk2::SourceView::Buffer->new(undef);
--        $browser->{file_view_svbuffer}->set_check_brackets(FALSE);
-+        $browser->{file_view_svbuffer} = Gtk2::SourceView2::Buffer->new(undef);
-+        $browser->{file_view_svbuffer}->set_highlight_matching_brackets(FALSE);
-         $browser->{file_view_svbuffer}->set_max_undo_levels(0);
-         $browser->{file_view_svbuffer}->begin_not_undoable_action();
-         $browser->{file_view_svlangmgr} =
--            Gtk2::SourceView::LanguagesManager->new();
--        $browser->{file_view_sv} = Gtk2::SourceView::View->
-+            Gtk2::SourceView2::LanguageManager->new();
-+        $browser->{file_view_sv} = Gtk2::SourceView2::View->
-             new_with_buffer($browser->{file_view_svbuffer});
-         $browser->{file_view_sv}->modify_font($mono_font);
-         $browser->{file_view_sv}->set_cursor_visible(FALSE);
-@@ -3326,7 +3326,7 @@ sub update_browser_state($$)
-             $browser->{file_view_svbuffer}->
-                 place_cursor($browser->{file_view_svbuffer}->get_start_iter());
-             $browser->{file_view_svbuffer}->set_text("");
--            $browser->{file_view_svbuffer}->set("highlight", FALSE);
-+            $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
-             set_label_value($browser->{file_name_value_label}, "");
-             set_label_value($browser->{file_author_value_label}, "");
-             set_label_value($browser->{last_update_value_label}, "");
-@@ -3451,7 +3451,7 @@ sub display_file($$)
-     $browser->{file_view_svbuffer}->
-         place_cursor($browser->{file_view_svbuffer}->get_start_iter());
-     $browser->{file_view_svbuffer}->set_text("");
--    $browser->{file_view_svbuffer}->set("highlight", FALSE);
-+    $browser->{file_view_svbuffer}->set_highlight_syntax(FALSE);
- 
-     # Get contents.
- 
-@@ -3555,9 +3555,12 @@ sub display_file($$)
-                          && $mime_details->{syntax_highlight}))
-                     && defined($lang =
-                                $browser->{file_view_svlangmgr}->
--                                   get_language_from_mime_type($mime_type)))
-+                                   guess_language($browser->
-+                                                      {file_being_viewed}->
-+                                                      {short_name},
-+                                                  $mime_type)))
-                 {
--                    $browser->{file_view_svbuffer}->set("highlight", TRUE);
-+                    $browser->{file_view_svbuffer}->set_highlight_syntax(TRUE);
-                     $browser->{file_view_svbuffer}->set_language($lang);
-                 }
- 
